### PR TITLE
replace MsgEthermint to MsgEthereumTx

### PR DESCRIPTION
**Pull request by giskook**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: app/ante/AccountSetupDecorator.go [hunk 1]
```diff
@@ -32,8 +32,8 @@ func (asd AccountSetupDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simulate
 	}
 
 	for _, msg := range msgs {
-		if msgEthermint, ok := msg.(evmtypes.MsgEthermint); ok {
-			setupAccount(asd.ak, ctx, msgEthermint.From)
+		if msgEthermint, ok := msg.(evmtypes.MsgEthereumTx); ok {
```
Comment (ilovers) [inline app/ante/AccountSetupDecorator.go]: rename msgEthermint

File: app/rpc/namespaces/eth/api.go [hunk 2]
```diff
@@ -1640,11 +1614,13 @@ func (api *PublicEthereumAPI) pendingMsgs() ([]sdk.Msg, error) {
 		pendingData := pendingTx.Input
 		nonce, _ := api.accountNonce(api.clientCtx, pendingTx.From, true)
 
-		msg := evmtypes.NewMsgEthermint(nonce, &pendingTo, sdk.NewIntFromBigInt(pendingValue), pendingGas,
-			sdk.NewIntFromBigInt(pendingGasPrice), pendingData, pendingFrom)
+		msg := evmtypes.NewMsgEthereumTx(nonce, pendingTx.To, pendingValue, pendingGas,
+			pendingGasPrice, pendingData)
+		msg.SetFrom(pendingTx.From.String())
```
Comment (KamiD) [inline app/rpc/namespaces/eth/api.go]: unnecessary to call SetFrom in here

Comment (giskook) [reply app/rpc/namespaces/eth/api.go]: got it

File: x/evm/handler_test.go [hunk 3]
```diff
@@ -154,65 +153,6 @@ func (suite *EvmTestSuite) TestHandleMsgEthereumTx() {
...
-		to   = sdk.AccAddress(secp256k1.GenPrivKey().PubKey().Address())
-	)
-
-	testCases := []struct {
-		msg      string
-		malleate func()
-		expPass  bool
-	}{
-		{
-			"passed",
```
Comment (giskook) [inline x/evm/handler_test.go:170]: TestHandleMsgEthereumTx "passed" case covered it

File: x/evm/handler_test.go [hunk 4]
```diff
@@ -154,65 +153,6 @@ func (suite *EvmTestSuite) TestHandleMsgEthereumTx() {
...
-		},
-		{
-			"invalid state transition",
-			func() {
-				tx = types.NewMsgEthermint(0, &to, sdk.NewInt(1), 100000, sdk.NewInt(2), []byte("test"), from)
-			},
-			false,
-		},
-		{
-			"invalid chain ID",
```
Comment (giskook) [inline x/evm/handler_test.go:185]: TestHandleMsgEthereumTx "invalid chain ID" case covered it

File: x/evm/handler_test.go [hunk 5]
```diff
@@ -154,65 +153,6 @@ func (suite *EvmTestSuite) TestHandleMsgEthereumTx() {
...
-		{
-			"passed",
-			func() {
-				tx = types.NewMsgEthermint(0, &to, sdk.NewInt(1), 100000, sdk.NewInt(2), []byte("test"), from)
-				suite.app.EvmKeeper.SetBalance(suite.ctx, ethcmn.BytesToAddress(from.Bytes()), big.NewInt(100))
-			},
-			true,
-		},
-		{
-			"invalid state transition",
```
Comment (giskook) [inline x/evm/handler_test.go:178]: need confirm

File: x/evm/handler_test.go [hunk 6]
```diff
@@ -760,96 +700,6 @@ func (suite *EvmTestSuite) TestEvmParamsAndContractDeploymentWhitelistControllin
 	}
 }
 
-func (suite *EvmTestSuite) TestEvmParamsAndContractDeploymentWhitelistControlling_MsgEthermint() {
```
Comment (giskook) [inline x/evm/handler_test.go:763]: coverd by TestEvmParamsAndContractDeploymentWhitelistControlling_MsgEthereumTx

File: x/evm/handler_test.go [hunk 7]
```diff
@@ -1072,84 +914,6 @@ func (suite *EvmContractBlockedListTestSuite) TestEvmParamsAndContractBlockedLis
 	}
 }
 
-func (suite *EvmContractBlockedListTestSuite) TestEvmParamsAndContractBlockedListControlling_MsgEthermint() {
```
Comment (giskook) [inline x/evm/handler_test.go:1075]: covered by TestEvmParamsAndContractBlockedListControlling_MsgEthereumTx

File: x/evm/types/msg_test.go [hunk 8]
```diff
@@ -23,75 +23,6 @@ import (
 	"github.com/okex/exchain/libs/tendermint/crypto/secp256k1"
 )
 
-func TestMsgEthermint(t *testing.T) {
```
Comment (giskook) [inline x/evm/types/msg_test.go:26]: covered by TestMsgEthereumTx

File: x/evm/types/msg_test.go [hunk 9]
```diff
@@ -23,75 +23,6 @@ import (
...
-	require.True(t, bytes.Equal(msg.GetSignBytes(), sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(msg))))
-	require.True(t, msg.GetSigners()[0].Equals(fromAddr))
-	require.Equal(t, *msg.To(), ethcmn.BytesToAddress(addr.Bytes()))
-
-	// clear recipient
-	msg.Recipient = nil
-	require.Nil(t, msg.To())
-}
-
-func TestMsgEthermintValidation(t *testing.T) {
```
Comment (giskook) [inline x/evm/types/msg_test.go:44]: partially covered by TestMsgEthereumTxValidation, had add amount is zero test case

File: x/evm/types/msg_test.go [hunk 10]
```diff
@@ -23,75 +23,6 @@ import (
...
-
-		if tc.expectPass {
-			require.Nil(t, msg.ValidateBasic(), "test: %v", i)
-		} else {
-			require.NotNil(t, msg.ValidateBasic(), "test: %v", i)
-		}
-	}
-}
-
-func TestMsgEthermintEncodingAndDecoding(t *testing.T) {
```
Comment (giskook) [inline x/evm/types/msg_test.go:73]: covered by TestMsgEthereumTx_Amino/TestMsgEthereumTxRLPDecode/TestMsgEthereumTxRLPEncode

File: x/evm/watcher/watcher_test.go [hunk 11]
```diff
@@ -204,53 +204,6 @@ func TestHandleMsgEthereumTx(t *testing.T) {
 	}
 }
 
-func TestMsgEthermintByWatcher(t *testing.T) {
```
Comment (giskook) [inline x/evm/watcher/watcher_test.go:207]: readd the ut

File: x/evm/handler_test.go (hunk 5 above)
Comment (giskook) [reply x/evm/handler_test.go:178]: debug with the old code path,it return error "insufficient balance for transfer", add a new "insufficient balance for transfer" case for TestHandleMsgEthereumTx

**Review by KamiD — APPROVED** [review]

**Pull request merged**